mass_mailing_switzerland/models/res_partner.py

```diff
@@ -129,61 +135,89 @@ def _compute_sponsored_child_fields(self):
-        for contact in self.filtered(lambda c: c.mailing_contact_ids):
+        for contact in self.filtered(lambda c: c.mass_mailing_contact_ids):
             if "opt_out" in vals and vals["opt_out"] != contact.opt_out:
-                contact.mailing_contact_ids.mapped("subscription_list_ids") \
+                contact.mass_mailing_contact_ids.mapped("subscription_list_ids") \
                     .with_context(opt_out_from_partner=True).write({
                     "opt_out": vals["opt_out"]})
             if "category_id" in vals:
-                contact.mailing_contact_ids.write({
+                contact.mass_mailing_contact_ids.write({
                     "tag_ids": vals["category_id"]
                 })
                 update_partner_ids.append(contact.id)
             if "email" in vals and not vals["email"] and not self.env.context.get("import_from_mailchimp"):
-                contact.mapped("mailing_contact_ids").filtered(
+                contact.mapped("mass_mailing_contact_ids").filtered(
                     lambda c: c.email == contact.email).unlink()
-            mailchimp_fields = contact.mailing_contact_ids.mapped(
+            mailchimp_fields = contact.mass_mailing_contact_ids.mapped(
                 "list_ids.mailchimp_list_id.merge_field_ids.field_id.name")
             mailchimp_fields += base_mailchimp_fields
             for f in mailchimp_fields:
                 if f in vals and vals[f] != getattr(contact, f):
                     update_partner_ids.append(contact.id)
                     break
         super().write(vals)
+        for contact in self.filtered(lambda c: c.mass_mailing_contact_ids):
+            for mass_mailing_contact in contact.mass_mailing_contact_ids:
+                if "lastname" in vals or "firstname" in vals:
+                    mass_mailing_contact.write({"merged_salutation": mass_mailing_contact
+                                               .compute_salutation()})
+                if "sponsored_child_ids" in vals:
+                    mass_mailing_contact.write({"merged_child": mass_mailing_contact
+                                               .compute_sponsored_child_fields()})
+        for partner in self.filtered(lambda c: not c.odoo_list_ids):
+            partner.write({"odoo_list_ids": [(4, self._default_odoo_list_ids().id)]})
+        if vals.get("email") and len(self.mailing_contact_ids) == 0:
+            for partner in self:
+                if not partner.odoo_list_ids:
+                    partner.write({"odoo_list_ids": [(3, self._default_odoo_list_ids())]})
+                if not partner.opt_out:
+                    vals["partner_id"] = partner.id
+                    vals["odoo_list_ids"] = partner.odoo_list_ids
+                    mass_mailing_contact = self.env["mail.mass_mailing.contact"].create(vals)
+                    mass_mailing_contact.name = self.firstname + " " + self.lastname
+                    mass_mailing_contact.action_export_to_mailchimp()
```

Comment:
self is a collection. this could lead to errors

name should be added in vals

export is already done on contact creation